pkg/diagnostics/component_monitoring.go

```diff
@@ -166,27 +166,41 @@ func (c *componentMetrics) Init(appID, namespace string) error {
 	c.enabled = true
 	c.namespace = namespace
 
+	// This is a gross hack that facilitates unit testing.  Once it's set here
+	// the rest of the unit tests in the diagnostics package consume the
+	// identical global view that was first set here because alphabetically this
+	// file was encountered first.
+	//
+	// NOTE: It might be more practical to explicitly run this check in each of
+	// the diagnostic package _test.go files where we call x.Init.
```

Comment:
This should actually say "it might make sense to run this check in each of the x.Init functions."

In normal runtime this would be set by InitMetrics.  In unit tests, this would be set by `component_monitoring_test.go` first.

Reply:
you could add a initGlobals (or something like it) function (with a sync.once probably) that all x.Init will call. That way we don't rely on import ordering.  I also imagine adding custom Size distribution config in the future.